R/scaling.R

```diff
@@ -0,0 +1,170 @@
+#' Reverse a numeric continuous scale
+#'
+#' Reverses a continuous scale into a new variable. A 5-1 scale thus turns into
+#' a 1-5 scale. Missing values are retained. This function really just subtracts
+#' each value from the maximum value which was incremented by one.
+#'
+#' @param data a [tibble][tibble::tibble-package] or a [tdcmm] model
+#' @param scale_var a numeric variable for which the scale gets reversed
+#' @param name the name of the new variable. By default, this is the same name
+#' as the `scale_var` but suffixed with `_rev`.
+#'
+#' @return a [tdcmm] model
+#' @export
+#' @family scaling
+#'
+#' @examples
+#' WoJ %>% reverse_scale(autonomy_emphasis)
+#' WoJ %>% reverse_scale(autonomy_emphasis, name = "my_reversed_variable")
+#'
+#' WoJ %>%
+#'   reverse_scale(autonomy_emphasis) %>%
+#'   tab_frequencies(autonomy_emphasis, autonomy_emphasis_rev)
+reverse_scale <- function(data, scale_var,
```

Comment:
Reversing based on the max has some bug potential especially for inexperienced, careless users (i.e., students):
- What happens if the scale has negative values (e.g., -2 to +2)?
- What happens if the variable has residual values identified through integers? 
- What happens if the data does not contain the (potential) scale max?

Maybe one could optionally pass the scale min and max as arguments (inferred from the data if not passed); we could then check whether all data are within those boundaries and calculate the `reverse_base` from the user-provided max.

Reply:
You're absolutely right, that was knitted with too hot of a needle (as they may say ...). More robust update incoming ...